File: TELERIK C# OOP/TELERIK-OOP-HW3/ArrayOfStudents/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ArrayOfStudents
{
    class Student
    {
        private string firstName;
        private string lastName;
        private int age;

        public string FirstName
        { 
            get{ return this.firstName; }
            set{ this.firstName = value;}
        }
        public string LastName
        {
            get { return this.lastName; }
            set { this.lastName = value; }        
        }
        public int Age
        {
            get { return this.age; }
            set { this.age = value; }
        }

        public Student(string firstName, string lastName)
        {
            this.firstName = firstName;
            this.lastName = lastName;
        }

        public Student(string firstName, string lastName, int age)
            : this(firstName, lastName)
        { 
            this.age = age;
        }

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.Append(this.firstName + " " + this.lastName + " " + this.age);
            return sb.ToString();
        }


        public static void FirstBeforeSecond(Student[] students)
        {
            var secondBeforeFirstSecondWay =
                students.
                Where(student => student.FirstName.CompareTo(student.LastName) < 0).
                Select(student => student);
            foreach (var item in secondBeforeFirstSecondWay)
            {
                Console.WriteLine(item.FirstName + " " + item.LastName);
            }
        }

    }
}

File: TELERIK-JS-Applications/TELERIK-JS-APLICATIONS-HW1/TELERIK-JS-APLICATIONS-HW1/scripts/objects/Student.js
define([], function () {
    var Student = (function () {

        function Student(firstName, lastName, age, mark) {
            this.firstName = firstName || 'Ivan';
            this.lastName = lastName || 'Ivanov';
            this.age = age || 21;
            this.mark = mark || 2;
        }

        Student.prototype.toString = function () {
            return this.firstName + '__' + this.lastName + '__' + this.age + '__' + this.mark;
        }

        return Student;
    }());
    return Student;
});
